### PR TITLE
Document custom strategy build secrets

### PR DESCRIPTION
**Pull request by csrwng**
Documents the secrets field for custom build strategies

### CODE REVIEW

**Comment by csrwng** [comment]
@bparees ptal

**Comment by bparees** [comment]
lgtm.  @adellape @CowboysFan ptal.

**Comment by adellape** [comment]
:+1:

File: dev_guide/builds.adoc [hunk 1]
```diff
@@ -320,6 +320,49 @@ accessible socket.  Set the `*exposeDockerSocket*` option to *true* to make this
...
+            "name": "secret1"
+          },
+          "mountPath": "/tmp/secret1" <2>
+        },
+        {
+          "secretSource": {
+            "name": "secret2"
+          },
+          "mountPath": "/tmp/secret2"
+        },
```
Comment (xiuwang) [inline dev_guide/builds.adoc]: It's better to remove this comma.

Comment (csrwng) [inline dev_guide/builds.adoc]: Thank you @xiuwang. I fixed it here #987

**Pull request merged**